test/cypress/support/commands.js

```diff
@@ -242,16 +242,19 @@ Cypress.Commands.add('giLogin', (username, {
     cy.getByDT('app').should('have.attr', 'data-ready', 'true')
 
     cy.window().its('sbp').then(sbp => {
-      const joinedGroupPromise = new Promise((resolve) => {
+      const joinedGroupPromise = new Promise((resolve, reject) => {
         if (firstLoginAfterJoinGroup) {
           const eventHandler = ({ groupContractID }) => {
             sbp('okTurtles.events/off', JOINED_GROUP, eventHandler)
-            const invervalId = setInterval(() => {
+            sbp('chelonia/contract/retain', groupContractID, { ephemeral: true }).then(() => new Promise(resolve => setTimeout(resolve, 1000))).then(() => {
```

Comment:
Similar issues: can you keep the formatting nice and readable, perhaps by moving the first `.then` to a new line?

Then, why is a manual timeout being added? We have a rather strict policy of avoiding those as much as possible since they are unreliable and also slow down the tests.

Finally, could you add a comment here explaining why we're now retaining the group for some reason?

Reply:
> can you keep the formatting nice and readable

Sure

> Then, why is a manual timeout being added?

Good catch. I don't have an answer now, which means that it likely is something I added during testing and forgot to remove. I agree that it shouldn't be there, and if it's really needed for some reason, it needs a comment.

> Finally, could you add a comment here explaining why we're now retaining the group for some reason?

Sure, but the same reason as for the previous comment applies, that it is the proper way to 'sync' a contract and do something with it.